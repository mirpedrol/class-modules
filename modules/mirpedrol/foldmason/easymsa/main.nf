process FOLDMASON_EASYMSA {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/06/067e6389ab95497b753ba1deabaa6acbce25b99c8cfcf39c06d5c1af42fd7751/data':
        'community.wave.seqera.io/library/foldmason_pigz:88809eb5649534b0' }"

    input:
    tuple val(meta), path(pdbs)


    output:
    tuple val(meta), path("${prefix}.aln.gz"), emit: alignment
    path("versions.yml")                     , emit: versions


    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    foldmason easy-msa \\
        ${pdbs} \\
        ${prefix} \\
        tmp \\
        $args \\
        --threads $task.cpus

    # Change name according to nf-class standard
    mv ${prefix}_3di.fa ${prefix}.aln

    pigz -p ${task.cpus} ${prefix}.aln

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        foldmason: \$( foldmason --version )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    // TODO nf-class: Update the output files to generate
    """
    touch ${prefix}.aln
    pigz ${prefix}.aln

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        foldmason: \$( foldmason --version )
    END_VERSIONS
    """
}
