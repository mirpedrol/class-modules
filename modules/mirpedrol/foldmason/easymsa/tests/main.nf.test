nextflow_process {

    name "Test Process FOLDMASON_EASYMSA"
    script "../main.nf"
    process "FOLDMASON_EASYMSA"

    tag "modules"
    tag "modules_mirpedrol"
    tag "foldmason"
    tag "foldmason/easymsa"
    tag "untar"

    setup {

        run("UNTAR") {
            script "../../../../../modules/mirpedrol/untar/main.nf"
                process {
                    """
                        archive = file("https://raw.githubusercontent.com/nf-core/test-datasets/multiplesequencealign/testdata/af2_structures/seatoxin-ref.tar.gz", checkIfExists: true)
                        input[0] = Channel.of(tuple([id:'test'], archive))
                    """
                }
        }
    }

    test("seatoxin") {

        when {
            process {
                """
                    input[0] = UNTAR.out.untar.map { meta,dir -> [[ id:'test' ], file(dir).listFiles().collect()]}
                """
            }
        }

        then {
            assertAll(
                { assert process.success  },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("seatoxin - stub") {
        options "-stub"

        when {
            process {
                """
                    input[0] = UNTAR.out.untar.map { meta,dir -> [meta, file(dir).listFiles().collect()]}
                """
            }
        }

        then {
            assertAll(
                { assert process.success  },
                { assert snapshot(process.out).match() }
            )
        }
    }

}