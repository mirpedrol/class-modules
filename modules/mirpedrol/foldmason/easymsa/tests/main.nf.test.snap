{
    "seatoxin - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.aln.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d50924112e0518ae66bcb59723b78ba9"
                ],
                "alignment": [
                    [
                        {
                            "id": "test"
                        },
                        "test.aln.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d50924112e0518ae66bcb59723b78ba9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-31T11:50:05.644555"
    },
    "seatoxin": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.aln.gz:md5,e7da437c6ddb5ced60ff7c49b147a65d"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d50924112e0518ae66bcb59723b78ba9"
                ],
                "alignment": [
                    [
                        {
                            "id": "test"
                        },
                        "test.aln.gz:md5,e7da437c6ddb5ced60ff7c49b147a65d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d50924112e0518ae66bcb59723b78ba9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-31T11:46:40.093716"
    }
}